zebra-chain/src/amount.rs

```diff
@@ -191,6 +191,60 @@ where
     }
 }
 
+impl<C> std::ops::Mul<u64> for Amount<C>
+where
+    C: Constraint,
+{
+    type Output = Result<Amount<C>>;
+
+    fn mul(self, rhs: u64) -> Self::Output {
+        let value = (self.0 as u64)
+            .checked_mul(rhs)
+            .ok_or(Error::AmountMulError)?;
+        value.try_into()
+    }
+}
+
+impl<C> std::ops::Mul<Amount<C>> for u64
+where
+    C: Constraint,
+{
+    type Output = Result<Amount<C>>;
+
+    fn mul(self, rhs: Amount<C>) -> Self::Output {
+        rhs.mul(self)
+    }
+}
```

Comment:
As discussed, let's only implement `Mul` and `Div` for `Amount<NonNegative>`:

```suggestion
impl std::ops::Mul<u64> for Amount<NonNegative>
{
    type Output = Result<Amount<NonNegative>>;

    fn mul(self, rhs: u64) -> Self::Output {
        let value = (self.0 as u64)
            .checked_mul(rhs)
            .ok_or(Error::MultiplicationOverflow { amount: self.0, multiplier: rhs })?;
        value.try_into()
    }
}

impl std::ops::Mul<Amount<NonNegative>> for u64
{
    type Output = Result<Amount<NonNegative>>;

    fn mul(self, rhs: Amount<NonNegative>) -> Self::Output {
        rhs.mul(self)
    }
}
```